core/app/models/spree/address.rb

```diff
@@ -11,7 +11,9 @@ class Address < Spree::Base
     belongs_to :country, class_name: "Spree::Country", optional: true
     belongs_to :state, class_name: "Spree::State", optional: true
 
-    validates :firstname, :address1, :city, :country_id, presence: true
+    validates :firstname, presence: true, if: -> { name.blank? || lastname? }
```

Comment:
Why checking `lastname` as well?

Reply:
The latest version of the validation is lightly different: https://github.com/solidusio/solidus/pull/3447/commits/93db0be1d74d5666145d6f0e7f6a052a98c06e28#diff-14b461f472ff6adbfd491ed5674e4e3aR252.
We need to check `lastname` too to avoid a corner case in which `name` is blank, `lastname` is present and someone tries to set `firstname` to blank.

Reply:
Ah, makes sense!